nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/impl/SimpleDocumentModel.java

```diff
@@ -67,17 +67,54 @@
 
     protected final Map<String, DataModel> dataModels = new HashMap<>();
 
-    protected Set<String> schemas;
+    protected final Set<String> schemas = new HashSet<>();
 
     protected final Map<String, Serializable> contextData = new HashMap<>();
 
     protected Path path;
 
     protected String type;
 
+    /**
+     * @see {@link #ofSchemas(List)}
+     * @see {@link #ofSchemas(String...)}
+     */
+    public SimpleDocumentModel() {
+        anySchema = true;
+    }
+
+    /**
+     * @deprecated since 11.1. Use {@link #ofSchemas(List)} instead.
+     */
+    @Deprecated(since = "11.1")
     public SimpleDocumentModel(List<String> schemas) {
-        this.schemas = new HashSet<>();
         anySchema = false;
```

Comment:
is there a raison to initialise the `anySchema`, as it is a primitif boolean which is false by default

Reply:
It's `final` so it needs initialization in the constructors.

Reply:
yep, i didn't pay attention of the `final` keyword